### PR TITLE
Union<T1, T2> Creation Performance

### PR DESCRIPTION
**Pull request by megafinz**
Improved Union<T1, T2> creation performance (10 times less allocations, 20+ times less CPU time).

If everything is OK I'll do the same for other Union types.

### CODE REVIEW

File: SuccincT/Unions/Union{T1,T2}.cs [hunk 1]
```diff
@@ -25,21 +21,11 @@ public Union(T2 value) : this(unit)
             Case = Variant.Case2;
         }
 
-        // ReSharper disable once UnusedParameter.Local - unit param used to 
+        // ReSharper disable once UnusedParameter.Local - unit param used to
         // prevent JSON serializer from using this constructor to create an invalid union.
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "_")]
         private Union(Unit _)
```
Comment (megafinz) [inline SuccincT/Unions/Union{T1,T2}.cs:27]: Is that constructor still needed?

**Comment by megafinz** [comment]
Not familiar with the CI build process, don't know what to fix : (

**Comment by DavidArno** [comment]
@megafinz,

No worries; I'll take a look and see if I can work it out.

**Comment by DavidArno** [comment]
I looked at the output from appveyor ... and don't understand what it's doing. It might be related to the fact that they've updated their VS2017 compilation process now that VS2017 is RTM. All SuccincT builds are going through the VS2017 compiler on appveyor at the moment. So I might need some config changes to make the builds work. I'm busy this weekend, but I'll take a look at it on Monday, if that's OK?

**Comment by megafinz** [comment]
No problem, I can't do anything useful anyway : )

**Comment by DavidArno** [comment]
OK, I made a change to the `appveyor.yml` file and pushed it, and all built OK. So I re-ran this PR on appveyor. It now is reporting errors about BenchmarkDotNet, Version=0.10.3.0 being missing. Is what you checked in just missing a nuget reference?

**Comment by megafinz** [comment]
I don't think so, package.config file is fine and project file references correct library from packages folder. 

I added a console application project in that commit, maybe AppVeyor needs some additional configuration to build it? Build log file shows that none of that console app's dependencies are found, so I guess package restore is not working for that console app.

**Comment by megafinz** [comment]
https://www.appveyor.com/blog/2014/03/18/about-nuget-package-restore/

That seems to help.

**Pull request merged**